File: 크레인인형뽑기게임/java/20221104/src/Solution.java
import java.util.Stack;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;

        Stack<Integer> stack = new Stack<>();

        for (int move : moves) {
            int j = move - 1;

            System.out.println("move >>>>" + move + " && j >>>>" + j);

            for (int i = 0; i < board.length; i++) {
                if (board[i][j] > 0) {
                    if (stack.isEmpty()) {
                        stack.push(0);
                    }

                    if (stack.peek() == board[i][j]) {
                        stack.pop();
                        answer += 2;
                        board[i][j] = 0;
                        break;
                    }

                    stack.push(board[i][j]);
                    board[i][j] = 0;
                    break;
                }
            }
        }

        return answer;
    }
}

File: 크레인인형뽑기게임/javascript/20221104/solution.js
export default function solution(board, moves) {
  let answer = 0;

  const stack = [0];

  moves.forEach((move) => {
    const j = move - 1;

    for (let i = 0; i < board.length; i += 1) {
      if (board[i][j] > 0) {
        if (stack[stack.length - 1] === board[i][j]) {
          stack.pop();
          answer += 2;
          board[i][j] = 0;
          break;
        }

        stack.push(board[i][j]);
        board[i][j] = 0;
        break;
      }
    }
  });

  return answer;
}
